.gitignore

```diff
@@ -4,6 +4,16 @@
 /node_modules
 /src/backend/node_modules
 
+# testing
+src/frontend/coverage
+
+# next.js
+src/frontend/.next
+src/frontend/out/
+src/frontend/build
+src/frontend/.vercel
+src/frontend/node_modules
```

Comment:
could we just add node_modules/ here, so that any node modules is covered and we don't have to go into it everytime one gets added